harness/determined/common/experimental/experiment.py

```diff
@@ -259,10 +259,19 @@ def get_with_offset(offset: int) -> bindings.v1GetExperimentCheckpointsResponse:
             sort_by = sb
             smaller_is_better = config.get("searcher", {}).get("smaller_is_better", True)
 
+        reverse = not smaller_is_better
+
         def key(ckpt: checkpoint.Checkpoint) -> Any:
             training = ckpt.training
             assert training
-            return training.validation_metrics["avgMetrics"][sort_by]
+            metric = training.validation_metrics.get("avgMetrics") or {}
+            metric = metric.get(sort_by)
```

Comment:
```python
metric = training.validation_metrics.get("avgMetrics", {}).get(sort_by)
```

Reply:
you can't do this, because "avgMetrics" is actually defined, it just has a value of `None`. meaning `.get` will return `None` and throw a `NullPointerException`

Reply:
ah, ok